app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/CustomOptions.php

```diff
@@ -924,7 +924,7 @@ protected function getPriceFieldConfig($sortOrder)
                         'addbeforePool' => $this->productOptionsPrice->prefixesToOptionArray(),
                         'sortOrder' => $sortOrder,
                         'validation' => [
-                            'validate-number' => true
+                            'validate-zero-or-greater' => true
```

Comment:
@geet07 as I see from https://github.com/magento/magento2/search?q=validate-zero-or-greater&unscoped_q=validate-zero-or-greater this validation rule is often used together with `validate-number` thus we can face with similar issues.

Why not fix `validate-number` validation rule instead so that it works correctly with separators?

Reply:
Hi @orlangur,

As per your comment, I have resolved the issue of validate-number validation. Please check and let me know if there are any other changes require.